shimmingtoolbox/optimizer/lsq_optimizer.py

```diff
@@ -38,12 +39,18 @@ def __init__(self, coils: ListCoil, unshimmed, affine, opt_criteria='mse', reg_f
         self._initial_guess_method = 'mean'
         self.initial_coefs = None
         self.reg_factor = reg_factor
+        self.w_signal_loss = w_signal_loss
+        self.epi_te = epi_te
         self.reg_factor_channel = np.array([max(np.abs(bound)) for bound in self.merged_bounds])
 
         lsq_residual_dict = {
             allowed_opt_criteria[0]: self._residuals_mse,
             allowed_opt_criteria[1]: self._residuals_mae,
-            allowed_opt_criteria[2]: self._residuals_std
+            allowed_opt_criteria[2]: self._residuals_std,
+            #############################################
+            ####### Yixin add the following code ########
```

Comment:
```suggestion

```

Reply:
Thanks! addresed this in https://github.com/shimming-toolbox/shimming-toolbox/commit/9c2ba526bef3781adfbd5b34d95d37b3e7f5c7da